tests/periph_timer_cli/main.c

```diff
@@ -0,0 +1,308 @@
+
+    gpio_t pin = _get_pin(argv[4], argv[5]);
+
+    gpio_init(pin, GPIO_OUT);
+
+    int res = timer_init(timer_dev, freq, cb, (void*)(intptr_t)pin);
+
+    return _print_cmd_result("timer_init", res == 0, res);
+}
+
+int _timer_set(int argc, char **argv, bool absolute)
+{
+    int res;
+    int timer_dev = _check_param(argc, argv, 5, 5,
+                                 "dev channel ticks gpio_port gpio_pin");
+
+    if (timer_dev == ARG_ERROR) {
+        return -1;
+    }
+
+    unsigned long chan = _get_num(argv[2]);
+    unsigned long timeout = _get_num(argv[3]);
+    gpio_t pin = _get_pin(argv[4], argv[5]);
+
+    mutex_lock(&cb_mutex);
+
+    if (absolute) {
+        gpio_toggle(pin);
+        res = timer_set_absolute(timer_dev, chan, timeout);
+    }
+    else {
+        gpio_toggle(pin);
+        res = timer_set(timer_dev, chan, timeout);
+    }
+
+    /* wait for unlock by cb */
+    mutex_lock(&cb_mutex);
+
+    /* reset mutex state */
+    mutex_unlock(&cb_mutex);
+    return res;
+}
+
+int cmd_timer_set(int argc, char **argv)
+{
+    int res = _timer_set(argc, argv, false);
+    return _print_cmd_result("timer_set", res == 1, res);
+}
+
+int cmd_timer_absolute(int argc, char **argv)
+{
+    int res = _timer_set(argc, argv, true);
+    return _print_cmd_result("timer_set_absolute", res == 1, res);
+}
+
+int cmd_timer_clear(int argc, char **argv)
+{
+    int timer_dev = _check_param(argc, argv, 2, 2, "dev channel");
+
+    if (timer_dev == ARG_ERROR) {
+        return -1;
+    }
+
+    unsigned long chan = _get_num(argv[2]);
+
+    int res = timer_clear(timer_dev, chan);
+
+    return _print_cmd_result("timer_clear", res == 1, res);
+}
+
+int cmd_timer_read(int argc, char **argv)
+{
+    int timer_dev = _check_param(argc, argv, 1, 1, "dev");
+
+    if (timer_dev == ARG_ERROR) {
+        return -1;
+    }
+
+    int res = timer_read(timer_dev);
+    return _print_cmd_result("timer_read", true, res);
+}
+
+int cmd_timer_start(int argc, char **argv)
+{
+    int timer_dev = _check_param(argc, argv, 1, 1, "dev");
+
+    if (timer_dev == ARG_ERROR) {
+        return -1;
+    }
+
+    timer_start(timer_dev);
+    puts("Success: timer_start()\n");
+    return 0;
+}
+
+int cmd_timer_stop(int argc, char **argv)
+{
+    int timer_dev = _check_param(argc, argv, 1, 1, "dev");
+
+    if (timer_dev == ARG_ERROR) {
+        return -1;
+    }
+
+    timer_stop(timer_dev);
+    puts("Success: timer_stop()\n");
+    return 0;
+}
+
+int cmd_get_metadata(int argc, char **argv)
+{
+    (void)argv;
+    (void)argc;
+
+    printf("Success: [%s, %s]\n", RIOT_BOARD, RIOT_APPLICATION);
+
+    return 0;
+}
+
+static const shell_command_t shell_commands[] = {
+    { "timer_read_bench", "execute multiple reads to determine overhead",
+      cmd_timer_bench_read },
+    { "timer_init", "init_timer", cmd_timer_init },
+    { "timer_set", "set timer to relative value", cmd_timer_set },
+    { "timer_absolute", "set timer to absolute value", cmd_timer_absolute },
```

Comment:
should be `timer_set_absolute` to match (current) API

Reply:
done